src/Helpers/Frontend/Shortcode.php

```diff
@@ -10,17 +10,15 @@ class Shortcode {
 	 * @since 2.7.0
 	 */
 	public static function getReceiptShortcodeFromConfirmationPage() {
-		$pattern = get_shortcode_regex();
 		$post    = get_post( give_get_option( 'success_page', 0 ) );
 		$content = $post->post_content;
 
-		if (
-			! empty( $content ) &&
-			preg_match_all( '/' . $pattern . '/s', $content, $matches ) &&
-			array_key_exists( 2, $matches ) &&
-			in_array( 'give_receipt', $matches[2], true )
-		) {
-			return $matches[0][0];
+		if ( ! empty( $content ) ) {
+			preg_match( '/\[give_receipt(.*?)]/', $content, $matches );
```

Comment:
Let's add the insensitive flag here to make sure that's not an issue.
```suggestion
			preg_match( '/\[give_receipt(.*?)]/i', $content, $matches );
```